### PR TITLE
Add Device IDs to distance sensors (rangefinders); add support for SF30/[B,C] units

### PR DESCRIPTION
**Pull request by JacobCrabill**
**Describe problem solved by this pull request**
Most sensors already have unique device IDs assigned to them; these IDs are primarily used to aid calibration of sensors, by having a unique ID that is essentially hardware-determined.  I find it useful even outside of calibration, however, such as when doing log analysis - by decoding the device ID you know exactly what sensor (driver) the data came from, how it was connected to the flight controller, etc.  This PR adds these device IDs for all distance sensors.

**Describe your solution**
One issue here was that the existing DeviceId / DeviceBusType structure did not have an entry for Serial ports.  So, I've added a DeviceBusType_SERIAL to that enum, and I've taken a (somewhat hacky) approach of using the NuttX serial port number as the "bus" number (specifically, taking the "x" of "ttySx" as the bus number).  I'm open to better suggestions here of how to do this.

This also requires changing the `uint8 id` field of the `distance_sensor` topic to `uint32 device_id` and adding `DRV_DIST_DEVTYPE_` entries for all distance sensors which did not already have one.

**Test data / coverage**
No testing just yet.  I have a few Lightware products to test with in the near future, but that's it.  Most of this work has been flying for nearly a year in a private fork, though.

**Other Notes**
I'm still planning to do a cleanup pass before saying this is good to go; I'll squash my commits once complete.  Just wanted to make this available for viewing sooner rather than later.

I've also added support for the Lightware SF30/B and SF30/C laser rangefinders over serial; see the attached log.

### CODE REVIEW

File: src/drivers/distance_sensor/srf05/SRF05.cpp [hunk 1]
```diff
@@ -48,8 +48,10 @@
 
 SRF05::SRF05(const uint8_t rotation) :
 	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::hp_default),
-	_px4_rangefinder(0 /* device id not yet used */, rotation)
+	_px4_rangefinder(0 /* no device type for GPIO input */, rotation)
```
Comment (dagar) [inline src/drivers/distance_sensor/srf05/SRF05.cpp:51]: Hmm, this is certainly a weird one.

**Review by LorenzMeier — APPROVED** [review]

**Comment by LorenzMeier** [comment]
Thanks! Could you rebase?

**Comment by JacobCrabill** [comment]
Log from some outdoor "bench testing": https://review.px4.io/plot_app?log=a1b5c73a-a37d-4995-b78d-bb2ab8e7a0b0

**Comment by LorenzMeier** [comment]
@JacobCrabill Would you mind checking the CI failures and rebase? It is not flash space but real breakage. Thanks for putting in all the hard work!

**Comment by JacobCrabill** [comment]
> @JacobCrabill Would you mind checking the CI failures and rebase? It is not flash space but real breakage. Thanks for putting in all the hard work!

Thanks for pointing that out - hard to know sometimes if the failures are "real" or not.  Should be an easy fix.

**Comment by mcsauder** [comment]
Render Engine the CI failure, not these changes.  Changes all look good to me.  I'll give a quick test on a sensor or two I have sitting around and get back with a review in a day or so.  Nice work @JacobCrabill !

**Comment by JacobCrabill** [comment]
Rebased again; perhaps I should re-open this PR using a public PX4 branch?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -680,12 +687,17 @@ MavlinkReceiver::handle_message_hil_optical_flow(mavlink_message_t *msg)
 	/* Use distance value for distance sensor topic */
 	distance_sensor_s d{};
 
+	device::Device::DeviceId device_id;
+	device_id.devid_s.bus = device::Device::DeviceBusType::DeviceBusType_SERIAL;
+	device_id.devid_s.devtype = DRV_DIST_DEVTYPE_MAVLINK;
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp:692]: I think it's also an open question if mavlink should be a "bus type", but this works for now.

**Review by dagar — APPROVED** [review]

File: src/modules/mavlink/mavlink_receiver.cpp (hunk 2 above)
Comment (JacobCrabill) [reply src/modules/mavlink/mavlink_receiver.cpp:692]: Can easily add that if desired

Comment (dagar) [reply src/modules/mavlink/mavlink_receiver.cpp:692]: I think it makes sense then we'll know how to decode the device id with meaningful mavlink metadata. FYI @CarlOlsson

Comment (JacobCrabill) [reply src/modules/mavlink/mavlink_receiver.cpp:692]: Ok, just created a branch for that.  PR coming next

**Pull request merged**